docs/extensions/storage/_index.md

```diff
@@ -8,39 +8,144 @@ geekdocFilePath: _index.md
+
+The dashed lines in the diagram indicate requests that are made to authenticate requests or lookup the storage provider:
+1. After authenticating a request the proxy may either use the CS3 `userprovider` or the accounts service to fetch the user information that will be minted into the `x-access-token`.
+2. The gateway will verify the JWT signature of the `x-access-token` or try to authenticate the request itself, eg. using a public link token.
+
+{{< hint warning >}}
+The bottom part is lighter because we will deprecate it in favor of using only the CS3 user and group providers after moving some account functionality into reva and glauth. The metadata storage is not registered in the reva gateway to prevent seperate user metadata necessary for running the service from data that is being served directly.
+{{< /hint >}}
+
+## Terminology
+
+Communication is hard. And clear communication is even harder. You may encounter the following terms throughout the documentation, in the code or when talking to other developers. Just keep in mind that whenever you hear or read *storage*, that term needs to be clarified, because on its own it is too vague. PR welcome.
+
+### Resources
+A *resource* is a logical concept. It can be have different [types](https://cs3org.github.io/cs3apis/#cs3.storage.provider.v1beta1.ResourceType)
+- an actual *file*
+- a *container*, eg. a folder or bucket
+- a *symlink*, or
+- a *reference* which can point to a resource in another *storage provider*
+
+### References
+
+A [*reference*](https://cs3org.github.io/cs3apis/#cs3.storage.provider.v1beta1.Reference) is a logical concept. It identifies a *resource* and consists of either
+- a *path* based reference, used to identify a *resource* in the *namespace* of a *storage provider*. It must start with a `/`.
+- an [*id* based reference](https://cs3org.github.io/cs3apis/#cs3.storage.provider.v1beta1.ResourceId), uniquely identifying a *resounce* in the *namespace* of a *storage provider*. It consists of a `storage provider id` and an `opaque id`. The `storage provider id` must NOT start with a `/`.
+
+{{< hint info >}}
+The `/` is important because currenty the static *storage registry* uses a map to look up which *storage provider* is responsible for the resource. Paths must be prefixed with `/` so there can be no collisions between paths and storage provider ids in the same map.
+{{< /hint >}}
+
+{{< hint warning >}}
+**Proposed Change**
+
+A *reference* is a logical concept. It identifies a *resource* and consists of
+a *root* and a *path* relative to that *root*. 
+A *root* is a `storage space id` and a `logical uuid` of a resource inside the *storage space*:
+```
+<storages_pace>:<logical_uuid>:<relative_path>
+`------------root------------´ `----path-----´
+```
+Both, *root* and *path*, are optional. The `storage space id` should be a uuid to prevent references from breaking when a *user* or *storage space* gets renamed. These are all valid references:
```

Comment:
```suggestion
*root* and *path* are both optional. The `storage space id` should be an uuid to prevent references from breaking when a *user* or *storage space* gets renamed. The following are all valid references:
```

Reply:
`be a uuid`, uuid starts with an umlaut and is pronounced in the same way.